shell/platform/android/io/flutter/plugin/common/StandardCodec.java

```diff
@@ -0,0 +1,334 @@
+package io.flutter.plugin.common;
+
+import java.io.ByteArrayOutputStream;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A {@link MessageCodec} using binary messages in Flutter standard encoding.
```

Comment:
Ok.  Can you write a short spec for the protocol so that folks who maintain this code in the future can understand the design?  Maybe an appropriate place for the spec is in a comment in the Dart implementation, which we can then link to from the Java and ObjC implementations?